src/renderer/store/actions.js

```diff
@@ -118,6 +118,57 @@ async function fetchCachedData(id, defaultValue) {
   return result;
 }
 
+async function fetchBlockHeaderByHash({ state, commit }, { blockHash, done, failed }) {
+  commit('startRequest', { identifier: 'fetchBlockHeaderByHash' });
+
+  // Check if the block is in memory
+  let blockHeader = _.get(state.blockDetails, blockHash);
+
+  if (!blockHeader || typeof blockHeader === 'function') {
+    const currentNetwork = network.getSelectedNetwork();
+
+    // Check if the block is in the pouchdb cache
+    const blockHeaderStorageKey = `blockheaders.${currentNetwork.net}.${blockHash}`;
+    try {
+      blockHeader = await fetchCachedData(blockHeaderStorageKey);
+    } catch (defaultVal) {
+      blockHeader = null;
+    }
+
+    if (!blockHeader) {
+      try {
+        blockHeader = await new Promise((resolve, reject) => {
+          const rpcClient = network.getRpcClient();
+          rpcClient.query({
+            method: 'getblockheader',
+            params: [blockHash, true],
+          })
+            .then((res) => {
+              db.upsert(blockHeaderStorageKey, JSON.stringify(res.result));
+              resolve(res.result);
+            })
+            .catch(e => reject(e));
```

Comment:
Is this indented correctly? It looks like it needs to go to the left once.

Reply:
it is correctly indented.